borsh-derive/src/lib.rs

```diff
@@ -59,8 +81,35 @@ pub fn borsh_deserialize(input: TokenStream) -> TokenStream {
     })
 }
 
+fn check_use_discriminant(derive_input: DeriveInput) -> Result<Option<bool>, TokenStream> {
```

Comment:
it's part of business logic and should be inside of `borsh_derive_internal` crate